nwbwidgets/timeseries.py

```diff
@@ -967,3 +972,349 @@ def update(
         fig.data = []
         fig.layout = {}
         return self.plot_group(group_inds,data,time_ts_aligned,fig,order)
+
+
+def trialize_time_series(
+    time_series: TimeSeries, trials_table: Union[DynamicTable, pd.DataFrame], index: int = 0
+) -> pd.DataFrame:
+
+    if isinstance(trials_table, DynamicTable):
+        trials_table_df = trials_table.to_dataframe()
+    else:
+        trials_table_df = trials_table
+
+    if time_series.timestamps:
+        timestamps = time_series.timestamps[:]
+    else:
+        number_of_frames = time_series.data.shape[0]
+        dt = 1.0 / time_series.rate
+        time_offset = time_series.starting_time if time_series.starting_time else 0
+        timestamps = np.arange(0, number_of_frames) * dt + time_offset
```

Comment:
use https://github.com/NeurodataWithoutBorders/nwbwidgets/blob/230d52e51828493ca141308d42aa91eebc92045d/nwbwidgets/utils/timeseries.py#L8-L41

Reply:
Done in the last commit.